clientlibs/js/src/ApiService/ApiService.ts

```diff
@@ -0,0 +1,318 @@
+      setGroupMemberShip: `${this.hostUrl}/api/${this.apiVersion}/groupmembership`,
+      setWorkingGroup: `${this.hostUrl}/api/${this.apiVersion}/workinggroup`,
+      failedExperimentPoint: `${this.hostUrl}/api/${this.apiVersion}/failed`,
+      getAllFeatureFlag: `${this.hostUrl}/api/${this.apiVersion}/featureflag`,
+      log: `${this.hostUrl}/api/${this.apiVersion}/log`,
+      logCaliper: `${this.hostUrl}/api/${this.apiVersion}/log/caliper`,
+      altUserIds: `${this.hostUrl}/api/${this.apiVersion}/useraliases`,
+      addMetrics: `${this.hostUrl}/api/${this.apiVersion}/metric`,
+    };
+    this.httpClient = this.setHttpClient(config.httpClient);
+  }
+
+  private setHttpClient(httpClient: UpGradeClientInterfaces.IHttpClientWrapper) {
+    if (USE_CUSTOM_HTTP_CLIENT && !httpClient) {
+      throw new Error(
+        'Please provide valid httpClient, or use the default (non-"lite") version of the library to our default httpClient.'
+      );
+    }
+
+    if (!USE_CUSTOM_HTTP_CLIENT && httpClient) {
+      throw new Error('Please import "lite" version of the to use custom httpClient.');
+    }
+
+    if (USE_CUSTOM_HTTP_CLIENT && httpClient) {
+      return httpClient;
+    }
+
+    if (!USE_CUSTOM_HTTP_CLIENT && !httpClient) {
+      return new DefaultHttpClient();
+    }
+  }
+
+  private validateClient() {
+    if (!USE_CUSTOM_HTTP_CLIENT && !this.hostUrl) {
+      throw new Error('Please set application host URL first.');
+    }
+    if (!this.userId) {
+      throw new Error('Please provide valid user id.');
+    }
+    if (!this.context) {
+      throw new Error('Please provide valid context.');
+    }
+    if (!this.httpClient) {
+      throw new Error('HttpClient is not defined.');
+    }
+  }
+
+  private createOptions(url: string): UpGradeClientInterfaces.IHttpClientWrapperRequestConfig {
+    const options: UpGradeClientInterfaces.IHttpClientWrapperRequestConfig = {
+      headers: {
+        'Content-Type': 'application/json',
+        'Session-Id': this.clientSessionId || uuid.v4(), // set this here? Do we need to require a clientSessionId?
+        'Client-source': IS_BROWSER ? 'browser' : 'node',
+        URL: url,
+      },
+    };
+
+    if (this.token) {
+      options.headers['Authorization'] = `Bearer ${this.token}`;
+    }
+
+    // layer in any custom headers supplied by the user
+    if (this.httpClient.config?.headers) {
+      options.headers = {
+        ...options.headers,
+        ...this.httpClient.config?.headers,
+      };
+    }
+
+    options.withCredentials = Boolean(this.httpClient.config?.withCredentials);
+
+    return options;
+  }
+
+  private async sendRequest<RequestBodyType>({ path, method, body }: IApiServiceRequestParams): Promise<any> {
+    this.validateClient();
+
+    const options = this.createOptions(path);
+
+    if (method === UpGradeClientEnums.REQUEST_METHOD.GET) {
+      return await this.httpClient.doGet(path, options);
```

Comment:
Why have you added await here? If you are not handling the error you should just return the promise.

Reply:
nice, I guess I thought it needed to be there. I was able to remove a ton of async/awaits in other places also that use this call.